frontend/src/app/game/game.service.ts

```diff
@@ -30,8 +35,20 @@ export class GameService {
   private _reset$ = new Subject<void>();
   reset$ = this._reset$.asObservable();
 
+  activateEvent$ = new Subject<ActivatedEvent | undefined>();
+
   constructor(private httpClient: HttpClient) {
     this.restartSimulation();
+
+    this.activateEvent$.pipe(
+      filter(activatedEvent => {
+        const mitigations = activatedEvent?.originEvent.mitigations;
+        return Boolean(mitigations && mitigations.length > 1);
+      }),
+      tap(activatedEvent => this.activatedEvent = activatedEvent),
+      debounceTime(this.PLAY_SPEED),
+      untilDestroyed(this),
+    ).subscribe(() => this.activatedEvent = undefined);
```

Comment:
This construct is pretty cryptic and error-prone, pls, could you improve it, pls?

Reply:
Idea is to once tick active event and remove it after "game speed" time passes. Any idea how to do it?

Reply:
totally removed and replaced with simple variable that is reset in `tick` method.